lambda/repository/lambda_functions.py

```diff
@@ -325,3 +445,44 @@ def presigned_url(event: dict, context: dict) -> dict:
 def get_groups(event: Any) -> List[str]:
     groups: List[str] = json.loads(event["requestContext"]["authorizer"]["groups"])
     return groups
+
+
+@api_wrapper
+def list_docs(event: dict, context: dict) -> list[RagDocument]:
+    """List all documents for a given repository/collection.
+
+    Args:
+        event (dict): The Lambda event object containing query parameters
+            - pathParameters.repositoryId: The repository id to list documents for
+            - queryStringParameters.collectionId: The collection id to list documents for
+        context (dict): The Lambda context object
+
+    Returns:
+        list[RagDocument]: A list of RagDocument objects representing all documents
+            in the specified collection
+
+    Raises:
+        KeyError: If collectionId is not provided in queryStringParameters
+    """
+
+    path_params = event.get("pathParameters", {})
+    repository_id = path_params.get("repositoryId")
+
+    query_string_params = event.get("queryStringParameters", {})
+    collection_id = query_string_params.get("collectionId")
+    docs: list[RagDocument] = doc_repo.list_all(repository_id, collection_id)
+
+    # return docs
+
+    id_token = get_id_token(event)
+    embeddings = _get_embeddings(model_name=collection_id, id_token=id_token)
+    vs = get_vector_store_client(repository_id, index=collection_id, embeddings=embeddings)
+
+    doc_search = vs.similarity_search(
```

Comment:
Do we really want to do a similarity search here to list out just the documents that have been loaded?

Reply:
This should have been removed. Good catch.